ngx_http_sxg_filter_module.c

```diff
@@ -663,15 +710,43 @@ static bool make_chain_from_buffer(ngx_http_request_t* req,
 
 static ngx_int_t ngx_http_sxg_body_filter(ngx_http_request_t* req,
                                           ngx_chain_t* in) {
-  ngx_http_sxg_srv_conf_t* ssc =
-      ngx_http_get_module_srv_conf(req, ngx_http_sxg_filter_module);
   ngx_http_sxg_ctx_t* ctx =
       ngx_http_get_module_ctx(req, ngx_http_sxg_filter_module);
 
-  if (ctx == NULL || req->done || req->header_sent) {
+  if (ctx == NULL || ctx->finished) {
```

Comment:
`ctx->finished` not protected by `ctx->lock`?

Reply:
`ctx->finished` is eliminated.